test/jdk/javax/swing/JRadioButton/8033699/bug8033699.java

```diff
@@ -57,64 +57,80 @@ public class bug8033699 {
-        robot.delay(100);
+        // Get all installed Look and Feels
+        UIManager.LookAndFeelInfo[] lafs = UIManager.getInstalledLookAndFeels();
+        for (UIManager.LookAndFeelInfo laf : lafs) {
+            testLaF(laf);
+        }
+    }
 
-        SwingUtilities.invokeAndWait(() -> mainFrame.dispose());
+    private static void testLaF(UIManager.LookAndFeelInfo laf) throws Exception {
+        try {
+            System.out.println("Testing LaF: " + laf.getName());
+            SwingUtilities.invokeAndWait(() -> {
+                setLookAndFeel(laf);
+                createAndShowGUI();
+            });
+
+            robot.waitForIdle();
+            robot.delay(1000);
+
+            // tab key test grouped radio button
+            runTest1();
+            robot.delay(100);
+
+            // tab key test non-grouped radio button
+            runTest2();
+            robot.delay(100);
+
+            // shift tab key test grouped and non-grouped radio button
+            runTest3();
+            robot.delay(100);
+
+            // left/up key test in grouped radio button
+            runTest4();
+            robot.delay(100);
+
+            // down/right key test in grouped radio button
+            runTest5();
+            robot.delay(100);
+
+            // tab from radio button in group to next component in the middle of button group layout
+            runTest6();
+            robot.delay(100);
+
+            // tab to radio button in group from component in the middle of button group layout
+            runTest7();
+            robot.delay(100);
+
+            // down key circle back to first button in grouped radio button
+            runTest8();
+            robot.delay(100);
+
+            // Verify that ActionListener is called when a RadioButton is selected using arrow key.
+            runTest9();
+            robot.delay(100);
+        } catch (Exception e) {
+            throw new RuntimeException("Error testing LaF: " + laf.getName(), e);
+        } finally {
+            SwingUtilities.invokeAndWait(() -> {
+                if (mainFrame != null) {
+                    mainFrame.dispose();
+                    mainFrame = null;
+                }
+            });
+        }
     }
 
-    private static void changeLAF() {
-        String currentLAF = UIManager.getLookAndFeel().toString();
-        System.out.println(currentLAF);
-        currentLAF = currentLAF.toLowerCase();
-        if (currentLAF.contains("nimbus")) {
-            try {
-                UIManager.setLookAndFeel("javax.swing.plaf.metal.MetalLookAndFeel");
-            } catch (Exception ex) {
-                ex.printStackTrace();
-            }
+    private static void setLookAndFeel(UIManager.LookAndFeelInfo laf) {
+        try {
+            UIManager.setLookAndFeel(laf.getClassName());
+        } catch (ClassNotFoundException | InstantiationException |
+                 IllegalAccessException | UnsupportedLookAndFeelException e) {
```

Comment:
This can just be set to `Exception e` rather than expanding it into separate exceptions.

Reply:
This provides more targeted error messages and logging.
If you catch a broad Exception, you might inadvertently mask other unexpected errors that should be investigated.
Separate exception handlers make it easier to identify the root cause of an issue.

Reply:
I'm fine with separate types.

You re-throwing any of the exceptions wrapped into `RuntimeException`. From this point of view, a single `Exception` won't change anything: if an exception is thrown, it's caught and re-thrown. These four exceptions are the only checked exceptions that can be thrown, catching `Exception` will catch them either way.

Either way is fine with me.